pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,477 @@
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p),
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+
+class ConduitNode:
+    def __init__(self, lib, ptr=None, child=False):
+        self.lib = lib
+        self.child = child
+
+        if ptr is not None:
+            self._as_parameter_ = ptr
+        else:
+            self._as_parameter_ = self.lib.conduit_node_create(None)
+
+    def __del__(self):
+        ptr = self._as_parameter_
+        if hasattr(self, 'lib') and ptr is not None and not self.child:
+            self.lib.conduit_node_destroy(self)
+        self._as_parameter_ = None
+
+    def __getitem__(self, key):
+        ptr = c_void_p(self.lib.conduit_node_fetch(self, key.encode()))
+        return ConduitNode(lib=self.lib, ptr=ptr)
+
+    def __setitem__(self, key, value):
+        match value:
+            case str():
+                self.lib.conduit_node_set_path_char8_str(self, key.encode(),
+                    value.encode())
+            case ConduitNode():
+                self.lib.conduit_node_set_path_node(self, key.encode(),
+                    value)
+            case int():
+                self.lib.conduit_node_set_path_int32(self, key.encode(),
+                    c_long(value))
+            case float():
+                self.lib.conduit_node_set_path_float32(self, key.encode(),
+                    c_float(value))
+            case (np.ndarray(dtype='float32') | np.generic(dtype='float32')):
+                vc = np.ascontiguousarray(value)
+                t, l = vc.dtype, c_longlong(vc.size)
+                ptr = vc.ctypes.data_as(POINTER(np.ctypeslib.as_ctypes_type(t)))
+
+                self.lib.conduit_node_set_path_float32_ptr(self, key.encode(),
+                    ptr, l)
+            case (np.ndarray(dtype='float64') | np.generic(dtype='float64')):
+                vc = np.ascontiguousarray(value)
+                t, l = vc.dtype, c_longlong(vc.size)
+                ptr = vc.ctypes.data_as(POINTER(np.ctypeslib.as_ctypes_type(t)))
+
+                self.lib.conduit_node_set_path_float64_ptr(self, key.encode(),
+                    ptr, l)
+            case (np.ndarray(dtype='int64') | np.generic(dtype='int64')):
+                vc = np.ascontiguousarray(value)
+                t, l = vc.dtype, c_longlong(vc.size)
+                ptr = vc.ctypes.data_as(POINTER(np.ctypeslib.as_ctypes_type(t)))
+
+                self.lib.conduit_node_set_path_int64_ptr(self, key.encode(),
+                    ptr, l)
+            case _:
+                ValueError(f'ConduitNode: __setitem__ type not supported')
+
+    def __str__(self):
```

Comment:
Not used.

Reply:
I use this frequently for debugging, I'll meet you half way, I'll remove the self.print and the detailed. It'll only add 5 lines and is very useful

Reply:
Right, but once code is debugged it stays that way.  Why does someone who downloads PyFR need debugging bloat which will never be called?

Reply:
Fine, I'll remove this in a final commit. I'm still using it in the process of responding to these requests

Reply:
Okay, I think we can remove.